### ISSUE TITLE
FinBERT Sentiment Analysis component for return estimation process

### ISSUE THREAD

**Issue by GongJr0** [GongJr0/Stock-Analyzer#7]
A sentiment analysis pipeline created using the libraries: `googlesearch`, `beautifulsoup4`, `transformers`, `pytorch`, `requests_cache`, and `sqlite3`

## Pipeline
1. Retrieve top $n$ results returned by a google search query (using `query = f"{text} after:{today() -timedelta(days=lookback_period)}`
2. Extract the paragraph and header text from the returned URLs using `bs4.find_all`
3. concatenate the text from each URL individually
4. Use the pre-trained FinBERT model and return a normalized sentiment polarity score for each URL. (score = $\frac{(S_{positive} - S_{negative})+1}{2}$)
5. Calculate the Exponentially Weighted Moving Average (EWMA) of the sentiment scores (recently posted URLs are more significant)
6. Return the EWMA value for further integration in the objective functions or expected return calculations
